napari/_qt/qt_dims.py

```diff
@@ -213,6 +233,11 @@ def _update_nsliders(self):
             if self._displayed_sliders[i]:
                 self._update_slider(i)
 
+    def _update_axis_labels(self, axis):
```

Comment:
This is one of the bigger issues I have, I'm not exactly sure what to write here, and it's a bit harder for me to "trial-and-error" it due to the larger issue I described in the conversation on this PR.

Reply:
This is where the new text from the dims model gets set to be the text on the corresponding QLineEdit box that gets displayed - something like
```python
        self.axis_labels[axis].setText(self.dims.axis_labels[axis])
```